crates/outline_panel/src/outline_panel.rs

```diff
@@ -4112,18 +4269,62 @@ impl OutlinePanel {
                     continue;
                 }
 
+                let mut last_depth_at_level: Vec<Option<Range<Anchor>>> = vec![None; 10]; // Track last outline at each depth level
+
                 for outline in excerpt.iter_outlines() {
-                    self.push_entry(
-                        state,
-                        track_matches,
-                        PanelEntry::Outline(OutlineEntry::Outline(OutlineEntryOutline {
-                            buffer_id,
-                            excerpt_id,
-                            outline: outline.clone(),
-                        })),
-                        outline_base_depth + outline.depth,
-                        cx,
-                    );
+                    let outline_entry = OutlineEntryOutline {
+                        buffer_id,
+                        excerpt_id,
+                        outline: outline.clone(),
+                    };
+
+                    // Check if this outline should be skipped due to a collapsed parent
```

Comment:
Could we move that into `iter_outlines`? (maybe add a "skip folded" param, but seems not needed also?)
Both places that call `iter_outlines` would benefit from this.